x/merkledb/proof.go

```diff
@@ -98,16 +95,16 @@ func (node *ProofNode) UnmarshalProto(pbNode *pb.ProofNode, bf BranchFactor) err
 	case pbNode.Key == nil:
 		return ErrNilKey
 	}
-	node.Key = ToKey(pbNode.Key.Value, bf).Take(int(pbNode.Key.Length))
+	node.Key = ToKey(pbNode.Key.Value).Take(int(pbNode.Key.Length))
 
-	if len(pbNode.Key.Value) != node.Key.bytesNeeded(node.Key.tokenLength) {
+	if len(node.Key.value) != bytesNeeded(node.Key.length) {
```

Comment:
I'm not sure I get this part. We are the ones setting `node.Key.value` and `node.Key.length` on line 98, so do we need to validate those values here?